nav2_costmap_2d/plugins/range_sensor_layer.cpp

```diff
@@ -0,0 +1,487 @@
+// Copyright 2018 David V. Lu!!
+// Ported to ROS2 by Michael Equi
+#include <angles/angles.h>
+#include <algorithm>
+#include <list>
+#include <limits>
+#include <string>
+#include <vector>
+
+#include "pluginlib/class_list_macros.hpp"
+#include "geometry_msgs/msg/point_stamped.hpp"
+#include "nav2_costmap_2d/range_sensor_layer.hpp"
+
+PLUGINLIB_EXPORT_CLASS(nav2_costmap_2d::RangeSensorLayer, nav2_costmap_2d::Layer)
+
+using nav2_costmap_2d::LETHAL_OBSTACLE;
+using nav2_costmap_2d::INSCRIBED_INFLATED_OBSTACLE;
+using nav2_costmap_2d::NO_INFORMATION;
+
+namespace nav2_costmap_2d
+{
+
+RangeSensorLayer::RangeSensorLayer() {}
+
+void RangeSensorLayer::onInitialize()
+{
+  current_ = true;
+  buffered_readings_ = 0;
+  last_reading_time_ = node_->get_clock()->now();
+  default_value_ = to_cost(0.5);
+
+  matchSize();
+  resetRange();
+
+  declareParameter("enabled", rclcpp::ParameterValue(true));
+  node_->get_parameter(name_ + "." + "enabled", enabled_);
+  declareParameter("phi", rclcpp::ParameterValue(1.2));
+  node_->get_parameter(name_ + "." + "phi", phi_v_);
+  declareParameter("inflate_cone", rclcpp::ParameterValue(1.0));
+  node_->get_parameter(name_ + "." + "phi", phi_v_);
+  declareParameter("no_readings_timeout", rclcpp::ParameterValue(0.0));
+  node_->get_parameter(name_ + "." + "no_readings_timeout", no_readings_timeout_);
+  declareParameter("clear_threshold", rclcpp::ParameterValue(0.2));
+  node_->get_parameter(name_ + "." + "clear_threshold", clear_threshold_);
+  declareParameter("mark_threshold", rclcpp::ParameterValue(0.8));
+  node_->get_parameter(name_ + "." + "mark_threshold", mark_threshold_);
+  declareParameter("clear_on_max_reading", rclcpp::ParameterValue(false));
+  node_->get_parameter(name_ + "." + "clear_on_max_reading", clear_on_max_reading_);
+
+  std::vector<std::string> topic_names{""};
```

Comment:
Do you really want a default of an empty vector? I'd think you'd rather have no default and then cause an exception if the user doesn't set this